### DISCUSSION TITLE
I Cannot handle the event of styled-component in React

### DISCUSSION THREAD

**Discussion by BOKYUNGGG** [styled-components/styled-components#3842] [Q&A] [answered]
Hello, my name is Bo Kyung.
I'm using React for a make portfolio.
I have some truble...

When I handle the event on styled-component, I can see the target's value, id, etc..
```
<StyledButton onClick={(e)=>{console.log(e.target.value)}} value="Styled Button">Styled Button</StyledButton >
```

Is this a problem with the react app or with the styled-component?

**Comment by auchynnikau** [comment]
@BOKYUNGGG  Try to change `target` to `currentTarget `

**Comment by quantizor** [comment, marked as answer]
This is a React behavior, not a styled-components issue. Use `e.currentTarget` instead of `e.target` to get the element the handler is attached to. `e.target` may refer to a child element that received the actual click.

styled-components forwards all event handlers and DOM props to the underlying element unchanged.
